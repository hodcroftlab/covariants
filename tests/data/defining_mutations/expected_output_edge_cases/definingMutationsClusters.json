{
  "clusters": [
    {
      "lineage": "B",
      "nextstrain_clade": "19A"
    },
    {
      "lineage": "B.1.1.515",
      "nextstrain_clade": "20B"
    },
    {
      "lineage": "XAW",
      "nextstrain_clade": "recombinant"
    },
    {
      "lineage": "LP.5",
      "nextstrain_clade": "24B"
    }
  ]
}