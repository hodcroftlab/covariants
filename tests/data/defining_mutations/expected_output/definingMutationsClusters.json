{
  "clusters": [
    {
      "pango_lineage": "B.1.1.7",
      "nextstrain_clade": "20I",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20B",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.351",
      "nextstrain_clade": "20H",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20C",
      "has_data": false
    },
    {
      "pango_lineage": "P.1",
      "nextstrain_clade": "20J",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20B",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.617.2",
      "nextstrain_clade": "21A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "21I",
        "21J"
      ],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": "BA.1",
      "nextstrain_clade": "21K",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "21M",
      "has_data": false
    },
    {
      "pango_lineage": "BA.2",
      "nextstrain_clade": "21L",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "23I",
        "22D",
        "22F",
        "22C",
        "22B",
        "22A",
        "24D"
      ],
      "nextstrain_parent": "21M",
      "has_data": false
    },
    {
      "pango_lineage": "BA.4",
      "nextstrain_clade": "22A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "21L",
      "has_data": false
    },
    {
      "pango_lineage": "BA.5",
      "nextstrain_clade": "22B",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "22E"
      ],
      "nextstrain_parent": "21L",
      "has_data": true
    },
    {
      "pango_lineage": "BA.2.12.1",
      "nextstrain_clade": "22C",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "21L",
      "has_data": false
    },
    {
      "pango_lineage": "BA.2.75",
      "nextstrain_clade": "22D",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "23C"
      ],
      "nextstrain_parent": "21L",
      "has_data": false
    },
    {
      "pango_lineage": "BQ.1",
      "nextstrain_clade": "22E",
      "pango_lineage_unaliased": "B.1.1.529.5.3.1.1.1.1.1",
      "pango_parent": "BE.1.1.1",
      "pango_children": [
        "BQ.1.1",
        "BQ.1.2",
        "BQ.1.3",
        "BQ.1.4",
        "BQ.1.5",
        "BQ.1.6",
        "BQ.1.7",
        "BQ.1.8",
        "BQ.1.9",
        "BQ.1.10",
        "BQ.1.11",
        "BQ.1.12",
        "BQ.1.13",
        "BQ.1.14",
        "BQ.1.15",
        "BQ.1.16",
        "BQ.1.17",
        "BQ.1.18",
        "BQ.1.19",
        "BQ.1.20",
        "BQ.1.21",
        "BQ.1.22",
        "BQ.1.23",
        "BQ.1.24",
        "BQ.1.25",
        "BQ.1.26",
        "BQ.1.27",
        "BQ.1.28",
        "BQ.1.29",
        "BQ.1.30",
        "BQ.1.31",
        "BQ.1.32",
        "BQ.1.33"
      ],
      "designation_date": "2022-09-02",
      "nextstrain_children": [],
      "nextstrain_parent": "22B",
      "has_data": true
    },
    {
      "pango_lineage": "BQ.1.1",
      "nextstrain_clade": "22E",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": null,
      "nextstrain_parent": null,
      "has_data": true
    },
    {
      "pango_lineage": "XBB",
      "nextstrain_clade": "22F",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "23A",
        "23B",
        "23D",
        "23E"
      ],
      "nextstrain_parent": "21L",
      "has_data": false
    },
    {
      "pango_lineage": "XBB.1.5",
      "nextstrain_clade": "23A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "23G"
      ],
      "nextstrain_parent": "22F",
      "has_data": false
    },
    {
      "pango_lineage": "XBB.1.16",
      "nextstrain_clade": "23B",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "22F",
      "has_data": false
    },
    {
      "pango_lineage": "CH.1.1",
      "nextstrain_clade": "23C",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "22D",
      "has_data": false
    },
    {
      "pango_lineage": "XBB.1.9",
      "nextstrain_clade": "23D",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "23F"
      ],
      "nextstrain_parent": "22F",
      "has_data": false
    },
    {
      "pango_lineage": "XBB.2.3",
      "nextstrain_clade": "23E",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "22F",
      "has_data": false
    },
    {
      "pango_lineage": "EG.5.1",
      "nextstrain_clade": "23F",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "23H"
      ],
      "nextstrain_parent": "23D",
      "has_data": false
    },
    {
      "pango_lineage": "XBB.1.5.70",
      "nextstrain_clade": "23G",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "23A",
      "has_data": false
    },
    {
      "pango_lineage": "HK.3",
      "nextstrain_clade": "23H",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "23F",
      "has_data": false
    },
    {
      "pango_lineage": "BA.2.86",
      "nextstrain_clade": "23I",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "24A"
      ],
      "nextstrain_parent": "21L",
      "has_data": false
    },
    {
      "pango_lineage": "JN.1",
      "nextstrain_clade": "24A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "24B",
        "24H",
        "24I"
      ],
      "nextstrain_parent": "23I",
      "has_data": false
    },
    {
      "pango_lineage": "JN.1.11.1",
      "nextstrain_clade": "24B",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "24C",
        "24G",
        "24F"
      ],
      "nextstrain_parent": "24A",
      "has_data": true
    },
    {
      "pango_lineage": "KP.3",
      "nextstrain_clade": "24C",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "24E"
      ],
      "nextstrain_parent": "24B",
      "has_data": false
    },
    {
      "pango_lineage": "XDV.1",
      "nextstrain_clade": "24D",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "21L",
      "has_data": false
    },
    {
      "pango_lineage": "KP.3.1.1",
      "nextstrain_clade": "24E",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "24C",
      "has_data": false
    },
    {
      "pango_lineage": "XEC",
      "nextstrain_clade": "24F",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "24B",
      "has_data": false
    },
    {
      "pango_lineage": "KP.2.3",
      "nextstrain_clade": "24G",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "24B",
      "has_data": false
    },
    {
      "pango_lineage": "LF.7",
      "nextstrain_clade": "24H",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "24A",
      "has_data": false
    },
    {
      "pango_lineage": "MV.1",
      "nextstrain_clade": "24I",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "24A",
      "has_data": false
    },
    {
      "pango_lineage": "LP.8.1",
      "nextstrain_clade": "25A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": null,
      "nextstrain_parent": null,
      "has_data": false
    },
    {
      "pango_lineage": "B.1.617.1",
      "nextstrain_clade": "21B",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.525",
      "nextstrain_clade": "21D",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.526",
      "nextstrain_clade": "21F",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20C",
      "has_data": false
    },
    {
      "pango_lineage": "C.37",
      "nextstrain_clade": "21G",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20D",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.621",
      "nextstrain_clade": "21H",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.177",
      "nextstrain_clade": "20E",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.427",
      "nextstrain_clade": "21C",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20C",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.429",
      "nextstrain_clade": "21C",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20C",
      "has_data": false
    },
    {
      "pango_lineage": null,
      "nextstrain_clade": "20C",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "21F",
        "21C",
        "20H",
        "20G"
      ],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": null,
      "nextstrain_clade": "21I",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "21A",
      "has_data": false
    },
    {
      "pango_lineage": null,
      "nextstrain_clade": "21J",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "21A",
      "has_data": false
    },
    {
      "pango_lineage": null,
      "nextstrain_clade": "21M",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "21K",
        "21L"
      ],
      "nextstrain_parent": "20B",
      "has_data": true
    },
    {
      "pango_lineage": "BE.1.1.1",
      "nextstrain_clade": "22B",
      "pango_lineage_unaliased": "B.1.1.529.5.3.1.1.1.1",
      "pango_parent": "BE.1.1",
      "pango_children": [
        "BQ.1",
        "BQ.2"
      ],
      "designation_date": "2022-08-24",
      "nextstrain_children": null,
      "nextstrain_parent": null,
      "has_data": true
    },
    {
      "pango_lineage": "BA.2.3.20",
      "nextstrain_clade": "21L",
      "pango_lineage_unaliased": "B.1.1.529.2.3.20",
      "pango_parent": "BA.2.3",
      "pango_children": [
        "CM.1",
        "CM.2",
        "CM.3",
        "CM.4",
        "CM.5",
        "CM.6",
        "CM.7",
        "CM.8",
        "CM.9",
        "CM.10",
        "CM.11",
        "CM.12"
      ],
      "designation_date": "2022-09-02",
      "nextstrain_children": null,
      "nextstrain_parent": null,
      "has_data": true
    },
    {
      "pango_lineage": "KP.2.3.8",
      "nextstrain_clade": "24G",
      "pango_lineage_unaliased": "B.1.1.529.2.86.1.1.11.1.2.3.8",
      "pango_parent": "KP.2.3",
      "pango_children": [],
      "designation_date": "2024-10-22",
      "nextstrain_children": null,
      "nextstrain_parent": null,
      "has_data": true
    },
    {
      "pango_lineage": "XBC",
      "nextstrain_clade": "recombinant",
      "pango_lineage_unaliased": "XBC",
      "pango_parent": null,
      "pango_children": [
        "XBC.1",
        "XBC.2"
      ],
      "designation_date": "2022-09-24",
      "nextstrain_children": null,
      "nextstrain_parent": null,
      "has_data": true
    },
    {
      "pango_lineage": "B",
      "nextstrain_clade": "19A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "20A",
        "19B"
      ],
      "nextstrain_parent": null,
      "has_data": false
    },
    {
      "pango_lineage": "B.1",
      "nextstrain_clade": "20A",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "20B",
        "21A",
        "20C",
        "21H",
        "21D",
        "21B",
        "20E"
      ],
      "nextstrain_parent": "19A",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.1",
      "nextstrain_clade": "20B",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "21M",
        "21E",
        "20J",
        "20I",
        "20F",
        "20D"
      ],
      "nextstrain_parent": "20A",
      "has_data": false
    },
    {
      "pango_lineage": "P.3",
      "nextstrain_clade": "21E",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20B",
      "has_data": false
    },
    {
      "pango_lineage": "D.2",
      "nextstrain_clade": "20F",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20B",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.1.1",
      "nextstrain_clade": "20D",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [
        "21G"
      ],
      "nextstrain_parent": "20B",
      "has_data": false
    },
    {
      "pango_lineage": "B.1.2",
      "nextstrain_clade": "20G",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "20C",
      "has_data": false
    },
    {
      "pango_lineage": "A",
      "nextstrain_clade": "19B",
      "pango_lineage_unaliased": null,
      "pango_parent": null,
      "pango_children": null,
      "designation_date": null,
      "nextstrain_children": [],
      "nextstrain_parent": "19A",
      "has_data": false
    }
  ]
}